p2p/background/kad_discovery_baseline_test.go

```diff
@@ -0,0 +1,128 @@
+package background
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	libp2pCrypto "github.com/libp2p/go-libp2p/core/crypto"
+	libp2pHost "github.com/libp2p/go-libp2p/core/host"
+	libp2pPeer "github.com/libp2p/go-libp2p/core/peer"
+	"github.com/multiformats/go-multiaddr"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/pocket/runtime/defaults"
+)
+
+func TestLibp2pKademliaPeerDiscovery(t *testing.T) {
+	ctx := context.Background()
+
+	addr1, host1, _ := setupHostAndDiscovery(t, ctx, defaults.DefaultP2PPort, nil)
+
+	bootstrapAddr, err := multiaddr.NewMultiaddr(fmt.Sprintf("%s/p2p/%s", addr1, host1.ID().String()))
+	require.NoError(t, err)
+
+	addr2, host2, _ := setupHostAndDiscovery(t, ctx, defaults.DefaultP2PPort+1, bootstrapAddr)
+	addr3, host3, _ := setupHostAndDiscovery(t, ctx, defaults.DefaultP2PPort+2, bootstrapAddr)
+
+	expectedPeerIDs := []libp2pPeer.ID{host1.ID(), host2.ID(), host3.ID()}
+
+	// TECHDEBT: consider using `host.ConnManager().Notifee()` to avoid sleeping here
+	// delay assertions for 500ms
+	time.Sleep(time.Millisecond * 500)
+
+	// assert that host2 has host3 in its peerstore
+	host2DiscoveredAddrs := host2.Peerstore().Addrs(host3.ID())
+	require.Lenf(t, host2DiscoveredAddrs, 1, "did not discover host3")
```

Comment:
This might be a huge NIT, but what if it's (somehow) it's greater than 1? Maybe we should update the error to be "expected exactly 1 address for host3 but found %d"

Reply:
Good catch! :+1: